iterator/snapshot.go

```diff
@@ -0,0 +1,174 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iterator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/conduitio-labs/conduit-connector-dynamodb/position"
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+// SnapshotIterator to iterate through DynamoDB items in a specific table.
+type SnapshotIterator struct {
+	tableName        string
+	partitionKey     string
+	sortKey          string
+	client           *dynamodb.Client
+	lastEvaluatedKey map[string]types.AttributeValue
+	items            []map[string]types.AttributeValue
+	firstIt          bool
+	index            int
+	p                position.Position
+}
+
+// NewSnapshotIterator initializes a SnapshotIterator starting from the provided position.
+func NewSnapshotIterator(tableName string, pKey string, sKey string, client *dynamodb.Client, p position.Position) (*SnapshotIterator, error) {
+	return &SnapshotIterator{
+		tableName:        tableName,
+		partitionKey:     pKey,
+		sortKey:          sKey,
+		client:           client,
+		lastEvaluatedKey: nil,
+		firstIt:          true,
+		p:                p,
+	}, nil
+}
+
+// todo can use dynamodb ScanPaginator instead
```

Comment:
Out of curiosity, what are the advantages of the ScanPaginator vs. ScanInput?

Reply:
nothing it's just simpler, I didn't know it existed before so i created my own way of pagination, doesn't have to be updated tbh it was just a note to myself that I forgot